Bug: calling non-function values passes type checking

## Description
Calling a non-function value (like an integer) passes type checking but fails at runtime.

## Reproduction
```ez
import @std
using std

do main() {
    temp x int = 5
    temp y = x()  // Should fail: calling non-function
    println(y)
}
```

Passes `ez check` but fails at runtime with:
```
error[E3015]: not a function: INTEGER
```

## Expected Behavior
The type checker should validate that call expressions are only used with function types.

--- a/integration-tests/fail/errors/E3015_call_non_function.ez
+++ b/integration-tests/fail/errors/E3015_call_non_function.ez
@@ -1,0 +1,12 @@
+/*
+ * Error Test: E3015 - call-non-function
+ * Expected: "cannot call" or "non-function"
+ */
+
+import @std
+using std
+
+do main() {
+    temp x int = 5
+    temp y = x()  // Should fail: calling non-function
+}

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -2007,6 +2007,20 @@ func (tc *TypeChecker) checkFunctionCall(call *ast.CallExpression) {
 	switch fn := call.Function.(type) {
 	case *ast.Label:
 		funcName = fn.Value
+		// Check if this is a variable being called as a function (#602)
+		if _, isVar := tc.lookupVariable(funcName); isVar {
+			// It's a variable, not a function - error
+			if _, isFunc := tc.functions[funcName]; !isFunc {
+				line, column := tc.getExpressionPosition(call.Function)
+				tc.addError(
+					errors.E3015,
+					fmt.Sprintf("cannot call non-function value '%s'", funcName),
+					line,
+					column,
+				)
+				return
+			}
+		}
 	case *ast.MemberExpression:
 		// Module function call - validate stdlib calls
 		tc.checkStdlibCall(fn, call)

